components/sensor/ultrasonic/ultrasonic.go

```diff
@@ -143,8 +143,10 @@ func (s *Sensor) Readings(ctx context.Context, extra map[string]interface{}) (ma
 		return nil, errors.Wrapf(err, "ultrasonic: cannot grab gpio %q", s.config.TriggerPin)
 	}
 
-	echoInterrupt.AddCallback(s.ticksChan)
-	defer echoInterrupt.RemoveCallback(s.ticksChan)
+	err = s.board.StreamTicks(ctx, []string{s.config.EchoInterrupt}, s.ticksChan, nil)
```

Comment:
Having this be variardic and passing in interrupt pins might be more user friendly, let's add that to the cleanup board interface scope doc.